File: src/ros2_bridge/mappers/foxglove_msgs/joint_state.rs
//! Typed mapper for `foxglove_msgs/msg/JointState`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn joint_state_to_bus(msg: ros_env::foxglove_msgs::msg::JointState) -> crate::foxglove_msgs::msg::v1::JointState {
    crate::foxglove_msgs::msg::v1::JointState {
        name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.name),
        position: msg.position,
        velocity: msg.velocity,
        acceleration: msg.acceleration,
        effort: msg.effort,
    }
}

pub(crate) fn joint_state_to_ros(bus: crate::foxglove_msgs::msg::v1::JointState) -> ros_env::foxglove_msgs::msg::JointState {
    ros_env::foxglove_msgs::msg::JointState {
        name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.name),
        position: bus.position,
        velocity: bus.velocity,
        acceleration: bus.acceleration,
        effort: bus.effort,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsJointStateMapper;

impl TypedTopicMapper for FoxgloveMsgsJointStateMapper {
    type Ros = ros_env::foxglove_msgs::msg::JointState;
    type Bus = crate::foxglove_msgs::msg::v1::JointState;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(joint_state_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(joint_state_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/joint_state.py
"""Generated mapper for `foxglove_msgs/msg/JointState`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.foxglove_msgs.msg.v1 import JointState as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def joint_state_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.name = str(msg.name)
    bus.position = msg.position
    bus.velocity = msg.velocity
    bus.acceleration = msg.acceleration
    bus.effort = msg.effort
    return bus


def joint_state_to_ros(bus):
    from foxglove_msgs.msg import JointState as RosMsg

    out = RosMsg()
    out.name = str(bus.name)
    out.position = bus.position
    out.velocity = bus.velocity
    out.acceleration = bus.acceleration
    out.effort = bus.effort
    return out


class FoxgloveMsgsJointStateMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from foxglove_msgs.msg import JointState as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return joint_state_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return joint_state_to_ros(bus)
